File: languages/cpp/apps/fib_test/fib.cxx
#include <iostream>
#include <cstdint>

uint64_t fib(uint64_t n) {
    if(n == 0 || n == 1) { return n; } else { return fib(n - 1) + fib(n - 2); }
}

int main() {
    uint64_t n = 42;
    std::cout << "Fib " << n << " is " << fib(n) << std::endl;

    return 0;
}

File: languages/go/apps/fib_test/fib.go
package main

import "fmt"

func fib(n uint64) uint64 {
	if n == 0 || n == 1 {
		return n
	} else {
		return fib(n-1) + fib(n-2)
	}
}

func main() {
	var n uint64 = 42
	fmt.Printf("Fib %d is %d", n, fib(n))
}
